src/main.rs

```diff
@@ -1,35 +1,690 @@
+    /// let wrapped_box = filled_box.wrap(
+    ///     Patterns::Polkadots,
+    ///     true,
+    ///     Some(tag)
+    /// );
+    /// assert_eq!(
+    ///     wrapped_box.read_tag(),
+    ///     "
+    ///     To: Bob,\
+    ///     From: Sally,\
+    ///     Message: Happy Cake Day!
+    ///     "
+    /// );
+    /// ```
+    pub fn read_tag(self) -> String {
+        match self.tag {
+            Some(tag) => tag.read(),
+            None => "There was no tag to read.".to_string(),
+        }
+    }
+}
+
+/// The `Patterns` enum represents various different Patterns that could be used by [`GiftWrap`].
+/// Though it is used by the `GiftWrap`, the `Patterns` enum can be used anywhere in Rust.
+#[derive(Debug, PartialEq)]
+pub enum Patterns {
+    Red,
+    Blue,
+    Green,
+    Paisley,
+    Polkadots,
+    Sparkles,
+    KraftPaper,
+    NewsPaper,
+    Cloth
+}
+
+/// The `GiftTag` struct represents a gift tag that could be included with a gift's wrapping. It is
+/// used by the `GiftWrap` struct to include a recipient, a sender, and a message. Though it is
+/// utilized by `GiftWrap`, this struct can be used anywhere in Rust.
+///
+/// `GiftTag` has the following parameters:
+/// * `recipient` which represents the recipient of the gift as a String.
+/// * `sender` which represents the sender of the gift as a String.
+/// * `message` which can be a message to be included with the gift as a String.
+///
+/// #Methods
+/// ##write(recipient, sender, message)
+/// You can write a new `GiftTag` with the [`GiftTag::write()`] method. Example:
+/// ```
+/// let tag = GiftTag::write(
+///      "Bob".to_string(),
+///      "Sally".to_string(),
+///      "Happy Cake Day!".to_string()
+/// );
+/// ```
+///
+/// ##read()
+/// You can read a `GiftTag` with the [`GiftTag::read()`] method. Example:
+/// ```
+/// assert_eq!(tag.read,
+/// "
+/// To: Bob,\
+/// From: Sally,\
+/// Message: Happy Cake Day!
+/// "
+/// );
+/// ```
+#[derive(Debug, PartialEq)]
+pub struct GiftTag {
+    recipient: String,
+    sender: String,
+    message: String
+}
+
+impl GiftTag {
+    /// The `write` method accepts three arguments as Strings (a recipient, a sender, and a message)
+    /// and returns a `GiftTag`.
+    ///
+    /// #Arguments
+    /// * `recipient` - Accepts a String that represents a gift's recipient (the person receiving the
+    /// gift).
+    /// * `sender` - Accepts a String that represents a gift's sender (the person who sent the gift).
+    /// * `message` - Accepts a string that represents a message to the recipient from the sender to
+    /// be included with the gift.
+    ///
+    /// #Returns
+    /// Returns a `GiftTag`.
+    ///
+    /// #Example
+    /// ```
+    /// let tag = GiftTag::write(
+    ///      "Bob".to_string(),
+    ///      "Sally".to_string(),
+    ///      "Happy Cake Day!".to_string()
+    /// );
+    /// ```
+    pub fn write(recipient: String, sender: String, message: String) -> GiftTag {
+        GiftTag {
```

Comment:
Its not necessary, but typical to have functions like these, which accept necessary fields and construct a struct, to be named `new`. This is not a keyword like it is in many other languages (where calling `new GiftTag()` has new as a language keyword), but its a common pattern, and one I am quick to look for when scanning through documentation.

Reply:
Completely renamed "write" with "new". I didn't bother keeping any sort of "write" function as that will only be duplication of functions without any added benefit.

Commit https://github.com/hypnoseal/giftbox/commit/57210035b6e06c2f664959ec7371fb629be6677b